op-batcher/batcher/channel.go

```diff
@@ -247,3 +248,34 @@ func (c *channel) OldestL2() eth.BlockID {
 func (s *channel) Close() {
 	s.channelBuilder.Close()
 }
+
+func (s *channel) updateDATypeAndRebuild(p ChannelConfigProvider) {
+	if len(s.pendingTransactions) != 0 {
+		// Channel has already started to be sent to L1
+		// We should not modify the DA type, since this can cause
+		// problems with transaction replacement / cancelling
+		return
+	}
+	usingBlobsInitially := s.cfg.UseBlobs
+	channelBytes := []byte{} // TODO use actual channel data
+	s.cfg = p.ChannelConfig(channelBytes)
+	if s.cfg.UseBlobs == usingBlobsInitially {
+		// No change to DA type, proceed as planned
+		return
+	}
+	// DA type has been changed, we need to rebuild the channel
+	newChannelBuilder, err := NewChannelBuilder(s.cfg, s.channelBuilder.rollupCfg, s.LatestL1Origin().Number)
+	if err != nil {
+		panic(err)
+		// TODO don't panic
+	}
+	for _, b := range s.channelBuilder.blocks {
+		newChannelBuilder.AddBlock(b)
+		// TODO error handling here.
+		// TODO I wonder if we want to be rebuilding at a layer above in the channel manager,
+		// because we could end up filling this channel and need to overflow into another one?
+	}
+
+	s.channelBuilder = newChannelBuilder
```

Comment:
We need to do the rebuilding on a higher level, at the channel manager level.

With the changes above, if we needed to switch DA type we would just end up with a mostly empty blob channel ( if going from calldata to blobs), or need multiple channels all of a sudden (if going from blobs to calldata).

Reply:
Better approach: if we detect a config change, we should:
* grab all of the blocks from existing channels
* throw away all of the existing channels
* requeue all of the blocks

Reply:
We could also cache some information about how full we made blobs last time around, and use this to help us estimate things more accurately when moving from calldata to blobs.

Typically we only fail to fill up blob txs because of channel timeouts / on chains with less activity. For example, we only put 2.4 blobs in the last tx. We can use this blob fullness ratio to help do the estimation.